app/hotels/src/map/MapScreen.js

```diff
@@ -0,0 +1,129 @@
+// @flow
+
+import * as React from 'react';
+import { StyleSheet, View } from 'react-native';
+import { createFragmentContainer, graphql } from 'react-relay';
+import idx from 'idx';
+
+import Map from './MapView';
+import HotelSwipeList from './HotelSwipeList';
+import type { MapScreen as MapScreenData } from './__generated__/MapScreen.graphql';
+
+type Props_ = {|
+  data: MapScreenData,
+|};
+
+type Props = Props & $Shape<Props_>;
+
+type State = {|
+  selectedHotelIndex: number,
+|};
+
+const styles = StyleSheet.create({
+  container: {
+    ...StyleSheet.absoluteFillObject,
+    justifyContent: 'flex-end',
+    alignItems: 'center',
+  },
+});
+
+class MapView extends React.Component<Props, State> {
+  state = {
+    selectedHotelIndex: 0,
+  };
+
+  componentWillReceiveProps = (nextProps: Props) => {
+    if (this.props.data !== nextProps.data) {
+      // TODO keep correct index if node with the same id is still present
+      this.setState({ selectedHotelIndex: 0 });
+    }
+  };
+
+  getHotels = () => {
+    return idx(this.props, _ => _.data.edges) || [];
+  };
+
+  selectMarker = (selectedHotelIndex: number) => {
+    if (this.state.selectedHotelIndex === selectedHotelIndex) {
+      return;
+    }
+
+    this.setState({ selectedHotelIndex });
+  };
+
+  onSelectMarker = (hotelId: string) => {
+    const hotels = this.getHotels();
+    const index = hotels.findIndex(
+      hotel => idx(hotel, _ => _.node.id) === hotelId,
+    );
+
+    if (index !== -1) {
+      this.selectMarker(index);
+    }
+  };
+
+  onSnapToItem = (index: number) => {
+    this.selectMarker(index);
+  };
+
+  render = () => {
+    const { selectedHotelIndex } = this.state;
+    const hotels = this.getHotels();
+
+    return (
+      <View style={styles.container}>
+        <Map
+          hotels={hotels}
+          selectedIndex={selectedHotelIndex}
+          onSelectMarker={this.onSelectMarker}
+        />
+        <HotelSwipeList
+          items={hotels}
+          selectedIndex={selectedHotelIndex}
+          onSnapToItem={this.onSnapToItem}
+        />
+      </View>
+    );
+  };
+}
+
+graphql`
+  fragment MapScreenHotel on HotelAvailability {
+    id
+    price {
+      amount
+      currency
+    }
+    hotel {
+      name
+      mainPhoto {
+        thumbnailUrl
+      }
+      rating {
+        stars
+      }
+      coordinates {
+        lat
+        lng
+      }
+      address {
+        street
+        city
+        zip
+      }
+    }
+  }
+`;
+
+export default createFragmentContainer(
+  MapView,
+  graphql`
+    fragment MapScreen on HotelAvailabilityConnection {
+      edges {
+        node {
+          ...MapScreenHotel @relay(mask: false)
```

Comment:
I cannot accept PR with `mask: false`. It's a red alert and it indicates inappropriate usage of Relay. It's not used in this project and it's used only 2x in BLKN (not necessary though). I tried to figure out why you need it here but it's still unclear - should be quite straightforward to use fragments in this situation. Can you please help me understand why did you decide to ignore Relay data masking here? Thanks... 🙂

Reply:
Or is it still under active development? I was under impression that you are almost finishing...

Reply:
fixed now 🙂 , cc @mrtnzlml